tools/test/ignoretool.py

```diff
@@ -29,33 +29,37 @@
 
 
 def package_directories(packages):
-    for directory in map(lambda p: 'lib/' + p.replace('.', '/'), packages):
-        yield directory
+    """Convert package names into directory paths under 'lib/'."""
+    return ['lib/' + p.replace('.', '/') for p in packages]
 
 
-def main():
-    parser = argparse.ArgumentParser(description='Returns ignore-strings for client files')
-    parser.add_argument('-l', '--pylint', dest='pylint', action='store_true',
-                        help='build a list for pylint with the --ignore flag')
-    parser.add_argument('-f', '--flake8', dest='flake8', action='store_true',
-                        help='build a list for flake8 tool')
-    script_args = parser.parse_args()
+def get_excluded_dirs():
+    """Find directories to exclude for Ruff, excluding __pycache__ and test files."""
+    include_dirs = set(package_directories(PACKAGES))
+    exclude_dirs = []
 
-    if script_args.flake8:
-        print(' '.join(map(lambda d: d + '/*.py', package_directories(PACKAGES))))
-    elif script_args.pylint:
-        ignore_dirs = []
-        include_dirs = set(package_directories(PACKAGES))
+    for root, dirs, files in os.walk('lib/'):
+        if '__pycache__' in root or root == 'lib/' or root in include_dirs:
+            continue
+        exclude_dirs.append(root)
 
-        for root, dirs, files in os.walk('lib/'):
-            if '__pycache__' not in root and root != 'lib/' and root not in include_dirs:
-                ignore_dirs.append(root)
```

Comment:
I'm confused as why this was changed - the behaviour seems to be the same

Reply:
I agree that the behavior is similar, but before, we were using `if not condition: do something` to check which directories to ignore. This made the logic a bit harder to follow. Now, we use `if condition: continue`, which immediately skips unwanted directories and only adds the valid ones. There’s no major performance difference, but this approach makes the code a little bit more cleaner and readable. Instead of checking what should be included, we now focus on filtering out what should be excluded first.